OJS.Workers.ExecutionStrategies/BlockchainStrategies/SolidityCompileDeployAndRunUnitTestsExecutionStrategy.cs

```diff
@@ -69,15 +69,21 @@ public class SolidityCompileDeployAndRunUnitTestsExecutionStrategy : ExecutionSt
 
         private IList<string> TestNames { get; } = new List<string>();
 
-        public override ExecutionResult Execute(ExecutionContext executionContext)
+        protected override IExecutionResult<TestResult> ExecuteAgainstTestsInput(
+            IExecutionContext<TestsInputModel> executionContext)
         {
-            var result = new ExecutionResult();
+            var result = new ExecutionResult<TestResult>();
 
-            this.ExtractTestNames(executionContext.Tests);
+            this.ExtractTestNames(executionContext.Input.Tests);
 
             // Compile the file
-            var compilerResult = this.ExecuteCompiling(executionContext, this.GetCompilerPathFunc, result);
-            if (!compilerResult.IsCompiledSuccessfully)
+            var isCompiledSuccessfully = this.ExecuteCompiling(
```

Comment:
I guess it will be too much refactoring, but ... IMO It will be better for the `ExecuteCompiling` to return `CompileResult` object or something, that contains compilation result (Successfull or Failed) and the compiler result.
This way, no `out` args will be used